### PR TITLE
Matrix divide by fix and improvements

### PR DESCRIPTION
**Pull request by congyu-lu**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/matrix.renderers.js [hunk 1]
```diff
@@ -324,16 +325,77 @@ export function setRenderers(self) {
...
+			.attr('class', 'sjpp-matrix-divide-by-label')
+			.attr('transform', `translate(0, ${y})`)
+
+		const textElem = g
+			.append('text')
+			.attr('text-anchor', anchor)
+			.attr('font-weight', 600)
+			.text(text)
+			.on('click', (event, d) => {
+				pill.showMenu(event, textElem.node())
```
Comment (xzhou82) [inline client/plots/matrix.renderers.js:356]: when the pill term is a categorical term with <= 2 categories e.g. Gender (count keys in term.values{}), then do not show Edit option, as is done when clicking term row label

Comment (congyu-lu) [reply client/plots/matrix.renderers.js:356]: pnet doens't show Edit option when categorical term <= 2 categories, but gdc matrix show Edit option for gender when clicking term row label or when clicking the **Group Cases By term** inside of cases control ui 
Screenshot 2023-09-20 at 17 21 15

Comment (xzhou82) [reply client/plots/matrix.renderers.js:356]: sorry didn't notice it's working for pnet. gdc is because the term.values{} lacks content. thanks!

**Review by xzhou82 — APPROVED** [review]
good job!

**Pull request merged**